packages/devextreme/js/__internal/ui/selection/m_selection.ts

```diff
@@ -43,109 +54,134 @@ export default class Selection {
-      filter: noop,
+      filter() { return undefined; },
     };
+    return defaultOptions;
   }
 
-  validate() {
+  validate(): void {
     this._selectionStrategy.validate();
   }
 
-  getSelectedItemKeys() {
-    return this._selectionStrategy.getSelectedItemKeys();
+  getSelectedItemKeys(): TDeferred extends true ? Promise<TKey[]> : TKey[] {
+    return this._selectionStrategy.getSelectedItemKeys() as TDeferred extends true
+      ? Promise<TKey[]>
+      : TKey[];
   }
 
-  getSelectedItems() {
-    return this._selectionStrategy.getSelectedItems();
+  getSelectedItems(): TDeferred extends true ? Promise<TItem[]> : TItem[] {
+    return this._selectionStrategy.getSelectedItems() as TDeferred extends true
+      ? Promise<TItem[]>
+      : TItem[];
   }
 
-  selectionFilter(value?: any) {
+  selectionFilter(value?: SelectionFilter): SelectionFilter | undefined {
     if (value === undefined) {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-return
       return this.options.selectionFilter;
     }
 
-    const filterIsChanged = this.options.selectionFilter !== value && JSON.stringify(this.options.selectionFilter) !== JSON.stringify(value);
+    const filterIsChanged = this.options.selectionFilter !== value
+      && JSON.stringify(this.options.selectionFilter) !== JSON.stringify(value);
 
     this.options.selectionFilter = value;
 
-    filterIsChanged && this.onSelectionChanged();
+    if (filterIsChanged) {
+      this.onSelectionChanged();
+    }
+
+    return undefined;
   }
 
-  setSelection(keys, updatedKeys?) {
+  setSelection(keys: TKey[], updatedKeys?: TKey[]): DeferredObj<TItem[]> {
     return this.selectedItemKeys(keys, false, false, false, updatedKeys);
   }
 
-  select(keys) {
+  select(keys: TKey[]): DeferredObj<TItem[]> {
     return this.selectedItemKeys(keys, true);
   }
 
-  deselect(keys) {
+  deselect(keys: TKey[]): DeferredObj<unknown> {
     return this.selectedItemKeys(keys, true, true);
   }
 
   selectedItemKeys(
-    keys,
+    keys: TKey[],
     preserve?: boolean,
     isDeselect?: boolean,
     isSelectAll?: boolean,
-    updatedKeys?: any[],
-  ) {
-    const that = this;
-
-    keys = keys ?? [];
-    keys = Array.isArray(keys) ? keys : [keys];
-    that.validate();
-
-    return this._selectionStrategy.selectedItemKeys(keys, preserve, isDeselect, isSelectAll, updatedKeys);
+    updatedKeys?: TKey[],
+  ): DeferredObj<TItem[]> {
+    let normalizedKeys = keys ?? [];
+    normalizedKeys = Array.isArray(normalizedKeys) ? normalizedKeys : [normalizedKeys];
+
+    this.validate();
+
+    return this._selectionStrategy.selectedItemKeys(
+      normalizedKeys,
+      preserve,
+      isDeselect,
+      isSelectAll,
+      updatedKeys,
+    );
   }
 
-  clearSelection() {
+  clearSelection(): DeferredObj<TItem[]> {
     return this.selectedItemKeys([]);
   }
 
-  _addSelectedItem(itemData, key) {
+  _addSelectedItem(itemData: TItem, key: TKey): void {
+    // @ts-expect-error addSelectedItem
```

Comment:
there type assertion will work
```suggestion
    (this._selectionStrategy as StandardStrategy).addSelectedItem(key, itemData);
```
better: create type guard and add check
```suggestion
      _isStandardStrategy(strategy: StandardStrategy | DeferredStrategy): strategy is StandardStrategy {
    return this.options.deferred;
  }

  _addSelectedItem(itemData: TItem, key: TKey): void {
    if (this._isStandardStrategy(this._selectionStrategy)) {
      this._selectionStrategy.addSelectedItem(key, itemData);
    }
  }
```

Reply:
this may change the actual behavior of this method so i suggested to explore this in the next issue (refactoring card has this as one of the items)